src/renderer_mtl.mm

```diff
@@ -2595,11 +2595,14 @@ void writeString(bx::WriterI* _writer, const char* _str)
 
 		if (NULL == _data)
 		{
-			m_ptr = s_renderMtl->m_device.newBufferWithLength(_size, 0);
```

Comment:
Wouldn't this opposite? Private when there is data?

Reply:
Mutable (NULL==data)
Update always uses blit command. (https://github.com/bkaradzic/bgfx/blob/master/src/renderer_mtl.mm#L2623
and 
https://github.com/bkaradzic/bgfx/blob/master/src/renderer_mtl.mm#L2629)
So a temporary system memory data is copied:
On discreet memory
Shared: Uses a memory area that is accessible by CPU and GPU. So faster copy but less optimal rendering.
Private: Uses video memory. Slower to copy, but optimal rendering.
Managed: There are two copies of the buffer (one in system and one in video memory). There is an extra copy from system memory to video memory inside metal driver.

So I think in this case private is the best. Because it uses video memory for rendering and avoids the double allocation and extra copy. 

Immutable case(data is not null):
You cannot set data for a private buffer. So an extra blit is needed from a shared buffer to a private. Managed does this copy automatically. But managed keeps the extra copy in system memory. So I think you are right.  It should be private with explicit copy.

I will change this to private and test it.

Reply:
Private for static buffers, Managed for dynamic buffers, and Shared is more for transient (not in current state, but with GPU markers to move put/get pointers).